2020_advent_of_code/day4/day4.cpp

```diff
@@ -0,0 +1,120 @@
+#include<iostream>
+#include<fstream>
+#include<vector>
+#include<unordered_map>
+#include<sstream>
+#include<regex>
+
+bool is_valid(const std::string& entry, const std::vector<std::regex>& regexes)
+{
+	std::smatch match;
+
+	for (int i = 0; i < regexes.size(); i++)
+	{
+		std::regex_match(entry, match, regexes[i]);
+		if (match.size() == 0) return false;
+	}
+
+	return true;
+}
+
+
+
+
+int main(int argc, char** argv)
+{
+	std::ifstream f(argv[1]);
+	std::string line;
+	std::vector<std::string> lines;
+
+	/*
+         *
+         * Regexes for applying rules
+         *
+         * 0 - byr must be in string
+         * 1 - iyr must be in string
+         * 2 - eyr must be in string
+         * 3 - hgt must be in string
+	 * 4 - hcl must be in string
+         * 5 - ecl must be in string
+         * 6 - pid must be in string
+         *
+         * Note:  cid is optional and not used for any validation
+         *
+         */
+        std::vector<std::regex> p1{
```

Comment:
Since these are known at compilation, probably should just use an array. Better yet for both of these vectors, see if you can reduce them to two regexes.